### PR TITLE
[Data][Docs] Add "Working with Text"

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

Text is a common data format for Data users. This PR adds a guide tailored specifically for this use case.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/working-with-text.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
+Working with Text
+=================
+
+Text data is ubiquitous in use cases like Natural Language Processing.
```
Comment (bveeramani) [inline doc/source/data/working-with-text.rst:4]: Could use recommendations on a better introductory sentence

Comment (amogkam) [reply doc/source/data/working-with-text.rst:4]: With Ray Data you can easily read and transform large datasets consisting of text data.

File: doc/source/data/working-with-text.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+This guide shows you how to:
+
+* :ref:`Read text files <reading-text-files>`
+* :ref:`Transform text <transforming-text>`
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:9]:
```suggestion
* :ref:`Transform text data <transforming-text>`
```

File: doc/source/data/working-with-text.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+* :ref:`Perform inference on text <performing-inference-on-text>`
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:10]:
```suggestion
* :ref:`Perform inference on text data <performing-inference-on-text>`
```

File: doc/source/data/working-with-text.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+* :ref:`Save text <saving-text>`
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:11]:
```suggestion
* :ref:`Save text data <saving-text>`
```

File: doc/source/data/working-with-text.rst [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+.. _reading-text-files:
+
+Reading text files
+------------------
+
+To read text files, call :func:`~ray.data.read_text`. Ray Data creates a row for each
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:18]: we also want to show read_binary_files here as well.

Lot's of text data are not cleanly stored in `.txt` files, and can be in `.pdf`, or `.html` or other file formats for example.

I don't think it's obvious `read_binary_files` should be used for these cases

File: doc/source/data/working-with-text.rst [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    {'text': 'explicit is better than implicit.'}
+
+For more information on transforming data, see `Transforming data <transforming_data>`_.
+
+.. _performing-inference-on-text:
+
+Performing inference on text
+----------------------------
+
+To perform inference on text, implement a callable class that sets up and invokes a
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:76]:
```suggestion
To perform inference with a pre-trained model on text data, implement a callable class that sets up and invokes a
```

File: doc/source/data/working-with-text.rst [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+.. testoutput::
+
+    {'text': 'The Zen of Python, by Tim Peters', 'label': 'POSITIVE'}
+    {'text': 'Beautiful is better than ugly.', 'label': 'POSITIVE'}
+    {'text': 'Explicit is better than implicit.', 'label': 'POSITIVE'}
+
+For more information on performing inference, see
+`End-to-end: Offline Batch Inference <batch_inference_home>`_
+and `Transforming batches with actors <transforming_data_actors>`_.
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:113]: these links don't seem to be working

File: doc/source/data/working-with-text.rst [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+    ds.show(3)
+
+.. testoutput::
+
+    {'text': 'the zen of python, by tim peters'}
+    {'text': 'beautiful is better than ugly.'}
+    {'text': 'explicit is better than implicit.'}
+
+For more information on transforming data, see `Transforming data <transforming_data>`_.
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:69]: link doesn't seem to work

File: doc/source/data/working-with-text.rst [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_text("s3://anonymous@ray-example-data/this.txt")
+
+    ds.write_parquet("local:///tmp/results")
+
+For more information on saving data, see `Saving data <saving_data>`_.
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:131]: link doesn't seem to work

File: doc/source/data/working-with-text.rst (hunk 5 above)
Comment (amogkam) [reply doc/source/data/working-with-text.rst:18]: even showing `read_json` could be useful. You get structured logs in json format

File: doc/source/data/working-with-text.rst [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+:ref:`End-to-end: Offline Batch Inference <batch_inference_home>`
+and :ref:`Transforming batches with actors <transforming_data_actors>`.
+
+.. _saving-text:
+
+Saving text
+-----------
+
+To save text, call a method like :meth:`~ray.data.Dataset.write_parquet`. Ray Data can
+save text in many formats.
```
Comment (amogkam) [inline doc/source/data/working-with-text.rst:177]: link to all the formats?

**Review by amogkam — APPROVED** [review]
nice work! I would just add some clarification on what JSONL is.

**Pull request merged**